src/imgui/imgui_raii_wrapper.h

```diff
@@ -0,0 +1,162 @@
+#ifndef MYPROJECT_IMGUI_RAII_WRAPPER_H
+#define MYPROJECT_IMGUI_RAII_WRAPPER_H
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <imgui.h>
+
+namespace irw {
+
+class Windowzita {
+public:
+  [[nodiscard]] explicit Windowzita(std::string const& id) : selected{ImGui::Begin(id.c_str())} {}
+  Windowzita(Windowzita const&) = delete;
+  Windowzita& operator=(Windowzita const&) = delete;
+  Windowzita(Windowzita&&) = delete;
+  Windowzita& operator=(Windowzita&&) = delete;
+  ~Windowzita() {
+    ImGui::End();
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabBar {
+public:
+  [[nodiscard]] explicit TabBar(std::string const& id) : selected{ImGui::BeginTabBar(id.c_str())} {}
+  TabBar(TabBar const&) = delete;
+  TabBar& operator=(TabBar const&) = delete;
+  TabBar(TabBar&&) = delete;
+  TabBar& operator=(TabBar&&) = delete;
+  ~TabBar() {
+    if (selected) {
+      ImGui::EndTabBar();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabItem {
+public:
+  [[nodiscard]] explicit TabItem(std::string const& id) : selected{ImGui::BeginTabItem(id.c_str())} {}
+  TabItem(TabItem const&) = delete;
+  TabItem& operator=(TabItem const&) = delete;
+  TabItem(TabItem&&) = delete;
+  TabItem& operator=(TabItem&&) = delete;
+  ~TabItem() {
+    if (selected) {
+      ImGui::EndTabItem();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class Text {
+public:
+  Text(char const* text_) : text{text_} {}
+  Text(std::string text_) : text{std::move(text_)} {}
+  Text(char const* text_, ImVec4 color_) : text{text_}, color{color_} {}
+  Text(std::string text_, ImVec4 color_) : text{std::move(text_)}, color{color_} {}
+  std::string get_text() const {
+    return text;
+  }
+  std::optional<ImVec4> get_color() const {
+    return color;
+  }
+  void write() const {
```

Comment:
Why use const here?

Reply:
`const` means constant. I suggest you research this a bit, because there are some uses for it. The general idea of `const` is that it is a promise we make that something won't be altered. In this specific case, we are promising that this member function will not alter the object (although we can break this promise, but let's not get into that now).